src/components/subscriptions/hooks/licenseManagerHooks.js

```diff
@@ -0,0 +1,208 @@
+      .then((response) => {
+        setIsLoading(false);
+        setSubscriptionUsers(response.data);
+      })
+      .catch((err) => {
+        NewRelicService.logAPIErrorResponse(err);
+        setIsLoading(false);
+        setError(err);
+      });
+  };
+
+  return {
+    fetch,
+    subscriptionUsers,
+    isLoading,
+    error,
+  };
+};
+
+/*
+This is a helper hook that provides all the subscription data in a single place.
+*/
+export const useSubscriptionData = ({ status, search, page = 1 }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [errors, setErrors] = useState(null);
+  const [data, setData] = useState({
+    subscriptions: {},
+    overview: {},
+    subscriptionUsers: {},
+  });
+
+  const {
+    fetch: fetchSubscriptions,
+    subscriptions,
+    isLoading: isLoadingSubscriptions,
+    error: subscriptionsError,
+  } = useSubscriptions();
+  const {
+    fetch: fetchSubscriptionUsersOverview,
+    subscriptionUsersOverview,
+    isLoading: isLoadingSubscriptionUsersOverview,
+    error: subscriptionUsersOverviewError,
+  } = useSubscriptionUsersOverview({ subscriptionUUID: subscriptions?.results[0]?.uuid });
+
+  const {
+    fetch: fetchSubscriptionUsers,
+    subscriptionUsers,
+    isLoading: isLoadingSubscriptionUsers,
+    error: subscriptionUsersError,
+  } = useSubscriptionUsers({
+    subscriptionUUID: subscriptions?.results[0]?.uuid,
+    options: { status, search, page },
+  });
+
+  const fetch = () => {
+    // This method will fetch subscriptions and after the successfully fetching subscriptions
+    // data, it will trigger the fetch operation of related data.
+    setIsLoading(true);
+    fetchSubscriptions();
+  };
+
+  useEffect(() => fetchSubscriptionUsersOverview(), [subscriptions]);
+  useEffect(
+    () => fetchSubscriptionUsers(),
+    [subscriptions, status, search, page],
+  );
+
+  useEffect(() => {
+    if (
+      isLoadingSubscriptions || isLoadingSubscriptionUsersOverview || isLoadingSubscriptionUsers
+    ) {
+      // Keep the status as isLoading until all requests complete.
+      // setIsLoading(true);
+    } else if (subscriptionsError || subscriptionUsersOverviewError || subscriptionUsersError) {
+      setIsLoading(false);
+      setErrors(prevState => ({
+        ...(prevState || {}),
+        ...(subscriptionsError ? { [SUBSCRIPTIONS]: NETWORK_ERROR_MESSAGE } : {}),
+        ...(subscriptionUsersError ? { [SUBSCRIPTION_USERS]: NETWORK_ERROR_MESSAGE } : {}),
+        ...(subscriptionUsersOverviewError ? {
+          [SUBSCRIPTION_USERS_OVERVIEW]: NETWORK_ERROR_MESSAGE,
+        } : {}),
+      }));
+    } else {
+      setErrors(null);
+      setIsLoading(false);
+      setData(prevState => ({
+        ...prevState,
+        subscriptions,
+        overview: subscriptionUsersOverview,
+        subscriptionUsers,
+      }));
+    }
+  }, [
+    subscriptions, isLoadingSubscriptions, subscriptionsError,
```

Comment:
Quite a long dependency list 😄 Curious if you think it makes sense to move some logic out into new variables to reduce the number of dependencies here, e.g.:

```
const isLoading = useMemo(
  () => isLoadingSubscriptions || isLoadingSubscriptionUsersOverview || isLoadingSubscriptionUsers,
 [isLoadingSubscriptions, isLoadingSubscriptionUsersOverview, isLoadingSubscriptionUsers],
);

useEffect(() => {
  if (isLoading) {
    // ...
  }
  return { //... };
}, [isLoading];
```